initial implementation of simple ft tutorial (#897)
near/docs: docs/tutorials/contracts/fts/simple-fts.md

![Wallet](/docs/assets/fts/empty-wallet-ft-tab.png)

## Using the FT contract

Now that you have all the tools in place, you're ready to start using the FT smart contract to [mint your FTs](#minting-your-fts).

:::note
If you don't have the command line interface, please [follow these steps](/docs/tools/near-cli#setup).
:::

Ensure you have credentials stored locally for the `testnet` account you will be using to mint tokens with by running the following `near-cli` command in your terminal:

```bash
near login
```

To make this tutorial easier to copy/paste, we're going to set an environment variable for your account ID. In the command below, replace `YOUR_ACCOUNT_NAME` with the account name you just logged in with including the `.testnet`):

```bash
export ID=YOUR_ACCOUNT_NAME
```

Test that the environment variable is set correctly by running:

```bash
echo $ID
```

### Minting your FTs

NEAR has already deployed a contract to the account `ft.examples.testnet` which allows anyone to freely mint `TeamTokens`. This is the account you'll be interacting with to mint your fungible tokens.

Now let's mint some tokens! The following command will mint `25` TeamTokens to your account.

> If you look carefully, the `receiver_id` defines the new owner of the tokens the user is minting, while `--accountId` sets the account that will be used to sign this call. 

Reviewer: ```suggestion
If you look carefully, the `receiver_id` defines the new owner of the tokens the user is minting, while `--accountId` sets the account that will be used to sign and pay for this transaction. 
```

> Now let's mint some tokens! The following command will mint `25` TeamTokens to your account.
> If you look carefully, the `receiver_id` defines the new owner of the tokens the user is minting, while `--accountId` sets the account that will be used to sign and pay for this transaction. 
> To front storage costs, the `--deposit` flag is set to attach `0.1` NEAR to the call.
> 
> Feel free to increase the `amount` up to a maximum value of 1000.

Reviewer: ```suggestion
Now let's mint some tokens! The command below will mint `25` TeamTokens to your account.
If you look carefully, the `receiver_id` defines the new owner of the tokens you are minting, while `--accountId` specifies which accounted will be used to sign and pay for this transaction. 
Also, there is a `--deposit` flag which attaches `.001` $NEAR to the call to [pay for storage](/docs/concepts/storage-staking) on the fungible token contract. The amount is slightly less than this but you will be refunded the difference. (See the transaction in your wallet) The amount of tokens to be minted is set to 25 but you can increase this value up to 1000.
```

---
id: simple-fts
title: Minting FTs (Simple)
sidebar_label: Minting FTs (Simple)
---

> In this tutorial you'll learn how to easily create your own fungible tokens without doing any software development by using a readily-available FT smart contract.

Reviewer 2: ```suggestion
In this tutorial, you'll learn how to easily create your own Fungible Tokens (FT) without doing any software development by using a readily-available FT smart contract.
```

## Overview

> This article will show you how to use an existing [FT smart contract](#fungible-token-contract), and you'll learn [how to mint](#minting-your-fts) fungible tokens and view them in your Wallet.

Reviewer 2: Fixed

### Wallet

To store your fungible tokens you'll need a [NEAR Wallet](https://wallet.testnet.near.org/).
If you don't have one yet, you can create one easily by following [these instructions](/docs/develop/basics/create-account):

> **Tip:** If you already have a NEAR `testnet` account, you can [skip these steps](#installing-the-near-cli).

#### 1. Reserve an Account ID

- Navigate to https://wallet.testnet.near.org and click on "Create Account".
- Next, enter your desired account name.

#### 2. Secure your account

- Choose your account recovery method. For simplicity, in this tutorial you can select
  "E-mail Account Recovery", although "Recovery Phrase" and [Ledger](https://www.ledger.com/)
  are the most secure methods.

#### 3. E-mail / Phone Number Account Recovery

- Enter the account activation code that you received.

#### 4. Success!

- You just created a `testnet` account and received 200 Ⓝ! Upon recovery method confirmation
  you should be directed to your account dashboard.

Once you have your Wallet account, you can click on the [Balances Tab](https://wallet.testnet.near.org/?tab=balances) where all your Fungible Tokens will be listed:

> ![Wallet](/docs/assets/fts/empty-wallet-ft-tab.png)

Reviewer: We don't need to have instructions on creating a wallet here. We can simply link to this process in the pre-req section. That way we avoid duplicate information in docs and eliminate the need to maintain multiple copies.

Reviewer 2: Fixed

interface, you can [skip these steps](#using-the-ft-contract).

> **Note:** Make sure you have a current version of `npm` and `NodeJS` installed.

#### Linux and macOS

1. Install `npm` and `node` using a package manager such as `nvm`. Sometimes there are issues
   using Ledger due to how macOS handles node packages related to USB devices.
   [[click here]](https://nodejs.org/en/download/package-manager/)
2. Ensure you have installed Node version 12 or above.
3. Install `near-cli` globally by running:

```bash
npm install -g near-cli
```

#### Windows

> **Note:** For Windows users, we recommend using Windows Subsystem for Linux (`WSL`).

1. Install `WSL` [[click here]](https://docs.microsoft.com/en-us/windows/wsl/install-manual#downloading-distros)
2. Install ` Node.js` [[click here]](https://nodejs.org/en/download/package-manager/)
3. Change `npm` default directory [[click here]](https://docs.npmjs.com/resolving-eacces-permissions-errors-when-installing-packages-globally#manually-change-npms-default-directory)
   - This is to avoid any permission issues with `WSL`
4. Open `WSL` and install `near-cli` globally by running:

```bash
npm install -g near-cli

> ```

Reviewer: Same here... we can remove this section and link to the CLI installation docs

Reviewer 2: Fixed

## Using the FT contract

Now that you have all the tools in place, you're ready to start using the FT smart contract to [mint your FTs](#minting-your-fts).

> Log in to your newly created account with `near-cli` by running the following command in your terminal.

Reviewer: ```suggestion
Ensure you have credentials stored locally for the `testnet` account you will be using to mint tokens with by running the following `near-cli` command in your terminal:
```

```bash
near login
```

> To make this tutorial easier to copy/paste, we're going to set an environment variable for your account ID. In the command below, replace `YOUR_ACCOUNT_NAME` with the account name you just logged in with including the `.testnet` (or `.near` for `mainnet`):

Reviewer: ```suggestion
To make this tutorial easier to copy/paste, we're going to set an environment variable for your account ID. In the command below, replace `YOUR_ACCOUNT_NAME` with the account name you just logged in with including the `.testnet`):
```

We can remove the mention of `.near` as this won't work on `mainnet`.

```bash
export ID=YOUR_ACCOUNT_NAME
```

Test that the environment variable is set correctly by running:

```bash
echo $ID
```

### Minting your FTs

> NEAR has already deployed a contract to the account `ft.examples.testnet` which allows users to freely mint `TeamTokens`. This is the account we'll be interacting with to mint our FTs.

Reviewer: ```suggestion
NEAR has already deployed a contract to the account `ft.examples.testnet` which allows anyone to freely mint `TeamTokens`. This is the account you'll be interacting with to mint your fungible tokens.
```

Best practice is to refer to the user by using "your" and "you'll" instead of "we" and "we'll"

> Now let's mint some tokens! The following command will mint 25 TeamTokens to your account. Feel free to increase the amount up to a max of 1000.

Reviewer: I think we should break down this command a little more. Explain that we are passing arguments and the `receiver_id` will be the new owner of the tokens the user is minting. More importantly, let's explain the need for the `--deposit flag` and the `--accountId flag`. Things like this are obvious to you and I but the new dev will not understand the need for these.

Reviewer 2: Added notes about required arguments/flags.

```bash
near call ft.examples.testnet ft_mint '{"receiver_id": "'$ID'", "amount": "25"}' --deposit 0.1 --accountId $ID
```

> To view tokens owned by an account you can call the FT contract with the following `near-cli` command:

Reviewer: ```suggestion
After minting your tokens you can query the contract to view the balance of any accountId passed. This is done by performing a contract `view` call and using the `ft_balance_of` endpoint on the fungible token contract. Run the following in your terminal to view the balance:
```

```bash
near view ft.examples.testnet ft_balance_of '{"account_id": "'$ID'"}'
```

<details>
<summary>Example response: </summary>
<p>

```json
View call: ft.examples.testnet.ft_balance_of({"account_id": "benji_test.testnet"})
'25'
```

</p>
</details>

> <br/>
>

> > **Tip:** after you mint your fungible token, you can [view it in your Wallet](https://wallet.testnet.near.org):

Reviewer: ```suggestion
:::tip
After you mint your fungible token you can [view it in your NEAR Wallet](https://wallet.testnet.near.org)!
:::
```
